### ISSUE TITLE
Site performance issues?

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#1230]
We are getting a small, but I think increasing, number of messages saying that iRecord is running slowly and/or not allowing various functions to complete (e.g. photo uploads failing). So far most of these messages have been quite vague, complaining about slowness in general rather than any particular issue.

I have found that I'm seeing pages fail to load more often than I was, at which point I get this message:
"The application did not respond in time."

Is there anything in the logs that would suggest we have a problem?

**Comment by BirenRathod** [comment]
@kitenetter I kind of anticipated to see if somebody else experiencing this problem. Back in December I experienced this problem while I was upgrading Drupal on iRecord. So yes there is a problem and on Pantheon the error stored as "Not enough PHP memory allocated." 

@DavidRoy I discussed with Pantheon and they suggest that iRecord site is running very busy. Look below the matrix. That suggest iRecord serves not less then 40,000 pages everyday. Always about more then 700 users are on websites.
![image](https://user-images.githubusercontent.com/5941255/148445539-94be4e96-7584-4fde-9cd1-1276e3263eba.png)

So Pantheon team recommend to go for Medium plan. Site is at present on basic plan. see below link for more detail about price plan.
![image](https://user-images.githubusercontent.com/5941255/148445818-aeac87b7-7a77-4fe3-a278-2dc5ff714c09.png)

**Comment by johnvanbreda** [comment]
I'll also have a look at the web-service stats on the warehouse this morning to see if there is anything that stands out.

@BirenRathod is it possible to determine which pages are being visited most and also which pages are taking the longest to serve or have this error occurring most?

**Comment by BirenRathod** [comment]
@johnvanbreda Here is the stats.

![image](https://user-images.githubusercontent.com/5941255/148525832-78346ac2-d637-4036-81f2-d7b54f6bde77.png)

**Comment by BirenRathod** [comment]
@johnvanbreda Also below is the average page load time.

![image](https://user-images.githubusercontent.com/5941255/148526717-cacd42c2-705a-409d-a718-6bbb6ff26192.png)

**Comment by johnvanbreda** [comment]
Thanks @BirenRathod. Looking at the stats, the casual record form seems a bit slow in particular. I suspect this is actually reflecting the time taken when saving a record rather than initial load - though I just tried it and it loaded and saved near-instantly.

I've now collected some stats on the warehouse and there is a report which populates the list of input forms on the filter builder. Although cached so it only runs occasionally, it was very slow so would probably occasionally make Pantheon think there was a timeout. I've just deployed a hotfix to change this report so it's much faster.

The next thing we will need to consider regarding warehouse performance is the save operation when adding records, which is now becoming one of the most significant consumers of warehouse time.
